test/Microsoft.ML.TimeSeries.Tests/TimeSeriesDirectApi.cs

```diff
@@ -571,6 +575,93 @@ public void AnomalyDetectionWithSrCnn(bool loadDataFromFile)
             }
         }
 
+        [Theory, CombinatorialData]
+        public void TestSrCnnBatchAnomalyDetector(
+            [CombinatorialValues(SrCnnDetectMode.AnomalyOnly, SrCnnDetectMode.AnomalyAndExpectedValue, SrCnnDetectMode.AnomalyAndMargin)]SrCnnDetectMode mode,
+            [CombinatorialValues(true, false)]bool loadDataFromFile)
+        {
+            var ml = new MLContext(1);
+            IDataView dataView;
+            if (loadDataFromFile)
```

Comment:
>loadDataFromFile [](start = 16, length = 16)

Why would there be a difference? Does that test different functionality of the new API?

Reply:
This is required by Harish to test both loading dataview from file and from List

---
In reply to: [426441364](https://github.com/dotnet/machinelearning/pull/5135#discussion_r426441364) [](ancestors = 426441364)

Reply:
In one of the earlier editions of the API (when it used some non-primitive data types) I thought I saw some issues with whether the API would support loading from files cleanly. So I had asked for this test.  It may not be required now. But if it is here already, it is is okay to leave it as is.

---
In reply to: [426707236](https://github.com/dotnet/machinelearning/pull/5135#discussion_r426707236) [](ancestors = 426707236,426441364)